src/dmd/expression.d

```diff
@@ -1353,7 +1353,7 @@ extern (C++) abstract class Expression : ASTNode
     extern (D) final bool checkReadModifyWrite(TOK rmwOp, Expression ex = null)
     {
         //printf("Expression::checkReadModifyWrite() %s %s", toChars(), ex ? ex.toChars() : "");
-        if (!type || !type.isShared())
+        if (!type || !type.isShared() || type.isTypeStruct() || type.isTypeClass())
```

Comment:
That doesn't look right. It's blindly allowing `struct` / `class`

Reply:
If the class/struct does not implement the operator the compiler will give the error "is not a scalar" (https://github.com/dlang/dmd/blob/d9826c828a8b228713f71136dc6108c42e295352/src/dmd/expressionsem.d#L5574)

So I don't think further checks are necessary